superchain/superchain-version_test.go

```diff
@@ -0,0 +1,90 @@
+package superchain_test
+
+import (
+	"context"
+	"fmt"
+	"reflect"
+	"testing"
+
+	"github.com/ethereum-optimism/superchain-registry/superchain"
+	"github.com/ethereum/go-ethereum"
+	"github.com/ethereum/go-ethereum/accounts/abi"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func checkErr(t *testing.T, err error) {
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestContractVersionsCheck will fail if the superchain semver file
+// is not read correctly.
+func TestSemverFile(t *testing.T) {
+	if err := superchain.SuperchainSemver.SanityCheck(); err != nil {
+		t.Fatal(err)
+	}
+}
+
+// TestContractVersionsCheck will check that
+//   - for each declared contract "Foo" in superchain.SuperchainSemver
+//   - for each chain in superchain.OPChains
+//
+// there is a contract address declared for "FooProxy" which has an
+// actual semver matching the declared semver. Actual semvers are
+// read from the L1 chain RPC provider for the chain in question.
+func TestContractVersionsCheck(t *testing.T) {
+	if err := superchain.SuperchainSemver.SanityCheck(); err != nil {
+		t.Fatal(err)
+	}
+
+	checkAllOPChainsSatisfySemver := func(contractName string) {
+		desiredSemver := reflect.ValueOf(superchain.SuperchainSemver).FieldByName(contractName).String()
+
+		// ASSUMPTION: we will check the version of the implementation via the declared proxy contract
+		proxyContractName := contractName + "Proxy"
+
+		for _, chain := range superchain.OPChains {
+			rpcEndpoint := superchain.Superchains[chain.Superchain].Config.L1.PublicRPC
+			t.Logf("Dialing %s...", rpcEndpoint)
```

Comment:
Right now we are blocked on RPC endpoints returning `400 Bad Request: {"error": "request has an invalid host info","id":1}`. I guess we need an API key to make this test reliable?

We may need to inject this manually at test time?

Reply:
Discussed on Slack: we'll use Alchemy or something to unblock for now, then figure out the better permanent solution

Reply:
I think the problem actually was mostly with the Goerli endpoint (I guess because it is deprecated). For now we can work apparently work around it using a different endpoint (tenderly) plus some retry logic so we don't anger their rate limiter.